# NEXUS — Supporting Analysis

This notebook contains calculations used to support methodological
decisions in the Executive Dashboard.

It is not the dashboard itself.

In [2]:
import pandas as pd
import numpy as np
import plotly.express as px

In [10]:
price_df = df.copy()

price_df = price_df[
    (price_df["SALES UNITS"] > 0) &
    (price_df["SALES PLN"] >= 0)
].copy()

price_df["ASP"] = (
    price_df["SALES PLN"] / price_df["SALES UNITS"]
)

price_df["ASP"].describe()

count    313265.000000
mean       4818.267783
std        3148.912487
min           0.000000
25%        2868.000000
50%        4084.000000
75%        5749.000000
max       88912.000000
Name: ASP, dtype: float64

Negative sales values are retained in the overall market analysis because they may represent sales adjustments or returns. For ASP and price-band analysis, only records with positive units and non-negative sales value are used, as these conditions are required to calculate a meaningful selling price.

In [ ]:
price_df = df.copy()

price_df = price_df[
    (price_df["SALES UNITS"] > 0) &
    (price_df["SALES PLN"] >= 0)
].copy()

price_df["ASP"] = (
    price_df["SALES PLN"] / price_df["SALES UNITS"]
)

price_df["ASP"].describe()

count    313265.000000
mean       4818.267783
std        3148.912487
min           0.000000
25%        2868.000000
50%        4084.000000
75%        5749.000000
max       88912.000000
Name: ASP, dtype: float64

In [ ]:
price_df["PRICE_BAND_CHECK"] = pd.cut(
    price_df["ASP"],
    bins=[0, 2500, 4000, 6000, float("inf")],
    labels=["0-2500", "2500-4000", "4000-6000", "6000+"]
)

price_band_check = (
    price_df
    .groupby("PRICE_BAND_CHECK", observed=False)["SALES UNITS"]
    .sum()
    .reset_index()D
)

price_band_check["SHARE"] = (
    price_band_check["SALES UNITS"] /
    price_band_check["SALES UNITS"].sum()
)

price_band_check

,PRICE_BAND_CHECK,SALES UNITS,SHARE
0,0-2500,1257924,0.262587
1,2500-4000,2150802,0.448971
2,4000-6000,1016962,0.212287
3,6000+,364821,0.076155


In [13]:
nexus_price = price_df[
    price_df["VENDOR"] == "NEXUS"
].copy()

nexus_price["PRICE_BAND_CHECK"] = pd.cut(
    nexus_price["ASP"],
    bins=[0, 2500, 4000, 6000, float("inf")],
    labels=["0-2500", "2500-4000", "4000-6000", "6000+"]
)

nexus_price_band = (
    nexus_price
    .groupby("PRICE_BAND_CHECK", observed=False)["SALES UNITS"]
    .sum()
    .reset_index()
)

nexus_price_band["SHARE"] = (
    nexus_price_band["SALES UNITS"] /
    nexus_price_band["SALES UNITS"].sum()
)

nexus_price_band

,PRICE_BAND_CHECK,SALES UNITS,SHARE
0,0-2500,275585,0.230450
1,2500-4000,605744,0.506537
2,4000-6000,241265,0.201751
3,6000+,73260,0.061262


## Price Band Decision

The price distribution was assessed for both the total market and NEXUS.

For the total market:
- 0–2500 PLN: 26.3% of units
- 2500–4000 PLN: 44.9% of units
- 4000–6000 PLN: 21.2% of units
- 6000+ PLN: 7.6% of units

For NEXUS:
- 0–2500 PLN: 23.1% of units
- 2500–4000 PLN: 50.7% of units
- 4000–6000 PLN: 20.2% of units
- 6000+ PLN: 6.1% of units

Based on the observed sales-volume distribution, the dashboard uses:

- Entry: < 2500 PLN
- Mainstream: 2500–4000 PLN
- Premium: > 4000 PLN

The 2500–4000 PLN range is the largest sales-volume segment for both the total market and NEXUS.